src/platform/monitoring/frontend-metrics/metrics.js

```diff
@@ -0,0 +1,166 @@
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/FormData
+ */
+// Exported for unit tests
+export function buildMetricsPayload(entry) {
+  const totalPageLoad = entry.duration;
+  const firstByte = entry.responseStart;
+  const domProcessing = entry.domComplete - entry.domInteractive;
+  const domComplete = entry.domComplete - entry.requestStart;
+  const domInteractive = entry.domInteractive;
+
+  let metrics = {
+    totalPageLoad,
+    firstByte,
+    domProcessing,
+    domComplete,
+    domInteractive,
+  };
+
+  if (contentfulPaintEntry()) {
+    const firstContentfulPaint = contentfulPaintEntry();
+    metrics = Object.assign({
+      firstContentfulPaint
+    }, metrics);
+  }
+  const metricsPayload = new FormData();
+  const metricsArray = [];
+  const pageUrl = window.location.pathname;
+
+  Object.keys(metrics).forEach(metric => {
+    metricsArray.push({
+      metric,
+      duration: metrics[metric]
+    });
+  });
+
+  // eslint-disable-next-line camelcase
+  const data = JSON.stringify({ metrics: metricsArray, page_id: pageUrl });
+  metricsPayload.append('data', data);
+
+  return metricsPayload;
+}
+
+/**
+ * Sends a payload via beacon. Asynchronously transmits data to the web server
+ * when the User-Agent has an opportunity to do so, without delaying the unload event
+ * or affecting the performance of the next navigation.
+ *
+ * @returns {boolean}
+ * @see: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon
+ */
+// Exported for unit tests
+export function sendMetricsToBackend(metricsPayload) {
+  const url = `${environment.API_URL}/v0/performance_monitorings`;
+  if (!navigator.sendBeacon(url, metricsPayload)) {
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Capture browser-provided navigation timing metrics from a PerformanceObserver object.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver/observe
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming
+ */
+// Exported for unit tests
+export function captureMetrics() {
+  const observer = new PerformanceObserver(list => {
+    list.getEntriesByType('navigation').forEach(entry => {
+      const metricsPayload = buildMetricsPayload(entry);
+      window.addEventListener('unload', () => {
+        sendMetricsToBackend(metricsPayload);
+      });
+    });
+  });
+
+  try {
+    observer.observe({ entryTypes: ['navigation'] });
+  } catch (error) {
+    Raven.captureException(error);
+  }
+}
+
+/**
+ * Check whether the browser supports the APIs we'll be using.
+ *
+ * @returns {boolean} Value is true if the current browser supports the PerformanceObserver object
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver/observe
+ */
+function canCaptureMetrics() {
+  if (performance === undefined) {
+    return false;
+  } else if (!('PerformanceObserver' in window)) {
+    return false;
+  }
+  return true;
+}
```

Comment:
Could this possibly be reduced to `return window.performance && window.PerformanceObserver`?